### PR TITLE
Admin jump overhaul, add watchlist functionality, other admin updates

### PR DESCRIPTION
**Pull request by Markolie**
Changes:
1) Most (JMP) commands have been replaced by (FLW) commands, allowing admins to instead follow the target.
2) Most log messages now make consistent use of JMP/FLW, key_name and key_name_admin.
3) It is now possible to add ckeys to a watchlist through the player panel. When these players join the server, a message will be sent to all admins (or IRC if there's no admins on): https://github.com/tgstation/-tg-station/pull/9556.
4) Admins that de-admin can now re-admin without having to reload admins: https://github.com/tgstation/-tg-station/pull/6743. Admins can now also hide all their verbs.
5) Added logging to various things, including players opening toxin/N2O canisters and flamethrowers.
6) Investigate logs now show proper timestamps.
7) Makes the morgue use Login/Logout hooks instead of snowflake code.
8) Antag HUD usage now properly checks for admin rights (R_MOD) instead of just checking if the client has a holder.
9) Added a config option to override the erro_ table prefix.
10) Mechas now destroy tables/racks when they hit them.

Fixes https://github.com/ParadiseSS13/Paradise/issues/1448.

And probably other fixes I forgot about.

DO NOT MERGE until the watchlist SQL changes are made, which will require earth's help.

### CODE REVIEW

**Comment by SamCroswell** [comment]
We already have 6.

**Comment by Markolie** [comment]
Updated.

**Comment by Fox-McCloud** [comment]
>8) Added logging to various things, including players opening toxin/N2O canisters.

THANK YOU. I was going to port this myself, but I kept forgetting.

**Comment by tigercat2000** [comment]
A number of these should really be in seperate commits, it's very difficult to sort through the changes with all the places JMP has been replaced.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -599,10 +599,10 @@
 		user << "\red You cannot locate any eyes on this creature!"
 		return
 
-	user.attack_log += "\[[time_stamp()]\]<font color='red'> Attacked [M.name] ([M.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
-	M.attack_log += "\[[time_stamp()]\]<font color='orange'> Attacked by [user.name] ([user.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
+	user.attack_log += "\[[time_stamp()]\]<font color='red'> Attacked [key_name(M)] with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
+	M.attack_log += "\[[time_stamp()]\]<font color='orange'> Attacked by [key_name(user)] with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
 	if(M.ckey)
-		msg_admin_attack("[user.name] ([user.ckey])[isAntag(user) ? "(ANTAG)" : ""] attacked [M.name] ([M.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)]) (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)") //BS12 EDIT ALG
```
Comment (tigercat2000) [inline code/game/objects/items.dm]: The isAntag() check here shouldn't be removed. adminmoreinfo crashes some admins, antag logs should be clearly marked out.

Comment (Markolie) [inline code/game/objects/items.dm]: isAntag checks are now handled by key_name_admin().

Comment (tigercat2000) [inline code/game/objects/items.dm]: Alright.

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -202,6 +203,48 @@ var/list/admin_verbs_mentor = list(
...
+		admin_verbs_sounds,
+		admin_verbs_spawn,
+		admin_verbs_mod,
+		admin_verbs_mentor,
+		admin_verbs_show_debug_verbs,
+		/client/proc/readmin,
+	)
+
+/client/proc/hide_verbs()
+	set name = "Adminverbs - Hide All"
```
Comment (tigercat2000) [inline code/modules/admin/admin_verbs.dm]: Um, this was removed a long time ago. No one used it. I am almost sure that Ponies and Neca told me not to readd it, when I inquired about it.

Comment (Markolie) [inline code/modules/admin/admin_verbs.dm]: Mhm, will ask around about it.

File: code/modules/admin/admin_verbs.dm [hunk 3]
```diff
@@ -526,12 +569,61 @@ var/list/admin_verbs_mentor = list(
...
+/client/proc/readmin()
+	set name = "Re-admin self"
+	set category = "Admin"
+	set desc = "Regain your admin powers."
+	var/datum/admins/D = admin_datums[ckey]
+	var/rank = null
+	if(config.admin_legacy_system)
+		//load text from file
+		var/list/Lines = file2list("config/admins.txt")
+		for(var/line in Lines)
```
Comment (tigercat2000) [inline code/modules/admin/admin_verbs.dm]: ... Copypasting the entire holder loader. Why didn't I think of that as a solution when I tried porting this. :unamused:

File: code/modules/mob/dead/observer/observer.dm [hunk 4]
```diff
@@ -598,4 +598,14 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		//add images for the 60inv things ghosts can normally see when darkness is enabled so they can see them now
 		client.images |= ghost_darkness_images
 		if (ghostimage)
-			client.images -= ghostimage //remove ourself
\ No newline at end of file
+			client.images -= ghostimage //remove ourself
+			
+//this is called when a ghost is drag clicked to something.
+/mob/dead/observer/MouseDrop(atom/over)
+	if(!usr || !over) return
```
Comment (tigercat2000) [inline code/modules/mob/dead/observer/observer.dm]: Errrmmmm... wasn't this handled... elsewhere...

File: config/example/dbconfig.txt [hunk 5]
```diff
@@ -20,6 +20,7 @@ PASSWORD mypassword
 FEEDBACK_DATABASE paradise
 FEEDBACK_LOGIN mylogin
 FEEDBACK_PASSWORD mypassword
+FEEDBACK_TABLEPREFIX myprefix
```
Comment (tigercat2000) [inline config/example/dbconfig.txt]: You should probably put a comment explaining what this is for.

**Comment by Necaladun** [comment]
Thankyouthankyouthankyou

File: code/__HELPERS/unsorted.dm [hunk 6]
```diff
@@ -653,7 +653,16 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
 
 /proc/key_name_admin(var/whom, var/include_name = 1)
-	return key_name(whom, 1, include_name)
+	var/message = "[key_name(whom, 1, include_name)](<A HREF='?_src_=holder;adminmoreinfo=\ref[whom]'>?</A>)[isAntag(whom) ? " (ANTAG)" : ""] ([admin_jump_link(whom, "holder")])"
+	return message
+	
+/proc/get_mob_by_ckey(key)
+	if(!key)
+		return
+	var/list/mobs = sortmobs()
```
Comment (Krausus) [inline code/__HELPERS/unsorted.dm]: I'm pretty sure that sorting `mob_list` here, just before searching it for a single mob by its key, does nothing but slow this proc down.

**Comment by Markolie** [comment]
SQL changes made.

**Comment by Fox-McCloud** [comment]
awesome; will merge as soon as this is ready.

**Pull request merged**